2021-09-11 16:20:08 [I] run.py:324 - Running command 'main' 
2021-09-11 16:20:08 [I] run.py:342 - Started run with ID "4" 
2021-09-11 16:20:09,798	INFO services.py:1247 -- View the Ray dashboard at [1m[32mhttp://127.0.0.1:8266[39m[22m
2021-09-11 16:20:11 [I] evolution.py:191 - Add  Subtract  Multiply  Div  Exp  Pow  Mean  LT  GE  Normalize  Max  Min 
2021-09-11 16:20:11 [I] evolution.py:196 - 0.0  1.0  1.5  2.0  3.0  0.1  0.01  0.001  -0.1  -0.01  -0.001 
2021-09-11 16:20:11 [I] evolution.py:205 - {'long_name': 'NFG-1', 'game_name': 'nfg_game', 'params': {'filename': 'nfg/NFG-1.nfg'}, 'transform': True, 'max_score': 1.1, 'weight': 1, 'iterations': 1000, 'name': 'NFG-1_1000'} 
2021-09-11 16:20:11 [I] evolution.py:205 - {'long_name': 'NFG-2', 'game_name': 'nfg_game', 'params': {'filename': 'nfg/NFG-2.nfg'}, 'transform': True, 'max_score': 1.1, 'weight': 1, 'iterations': 1000, 'name': 'NFG-2_1000'} 
2021-09-11 16:20:11 [I] evolution.py:205 - {'long_name': 'NFG-3', 'game_name': 'nfg_game', 'params': {'filename': 'nfg/NFG-3.nfg'}, 'transform': True, 'max_score': 1.1, 'weight': 1, 'iterations': 1000, 'name': 'NFG-3_1000'} 
2021-09-11 16:20:11 [I] evolution.py:205 - {'long_name': 'NFG-4', 'game_name': 'nfg_game', 'params': {'filename': 'nfg/NFG-4.nfg'}, 'transform': True, 'max_score': 1.1, 'weight': 1, 'iterations': 1000, 'name': 'NFG-4_1000'} 
2021-09-11 16:20:11 [I] evolution.py:225 - start reading algorithms from file and evaluating agents. 
2021-09-11 16:20:12 [I] evolution.py:304 - Agent 0 obtain score NFG-1_1000 in Game 0.0 
2021-09-11 16:20:12 [I] evolution.py:304 - Agent 1 obtain score NFG-1_1000 in Game 0.7889912971022862 
2021-09-11 16:20:12 [I] evolution.py:304 - Agent 2 obtain score NFG-1_1000 in Game 1.0 
2021-09-11 16:20:12 [I] evolution.py:304 - Agent 3 obtain score NFG-1_1000 in Game 1.1 
2021-09-11 16:20:12 [I] evolution.py:304 - Agent 0 obtain score NFG-2_1000 in Game 0.0 
2021-09-11 16:20:12 [I] evolution.py:304 - Agent 1 obtain score NFG-2_1000 in Game 0.7456676977322342 
2021-09-11 16:20:12 [I] evolution.py:304 - Agent 2 obtain score NFG-2_1000 in Game 1.0 
2021-09-11 16:20:12 [I] evolution.py:304 - Agent 3 obtain score NFG-2_1000 in Game 1.1 
2021-09-11 16:20:12 [I] evolution.py:304 - Agent 0 obtain score NFG-3_1000 in Game 0.0 
2021-09-11 16:20:12 [I] evolution.py:304 - Agent 1 obtain score NFG-3_1000 in Game 0.4434633568047555 
2021-09-11 16:20:12 [I] evolution.py:304 - Agent 2 obtain score NFG-3_1000 in Game 1.0 
2021-09-11 16:20:12 [I] evolution.py:304 - Agent 3 obtain score NFG-3_1000 in Game 0.5156807220795897 
2021-09-11 16:20:12 [I] evolution.py:304 - Agent 0 obtain score NFG-4_1000 in Game 0.0 
2021-09-11 16:20:12 [I] evolution.py:304 - Agent 1 obtain score NFG-4_1000 in Game 0.4278277300515892 
2021-09-11 16:20:12 [I] evolution.py:304 - Agent 2 obtain score NFG-4_1000 in Game 1.0 
2021-09-11 16:20:13 [I] evolution.py:304 - Agent 3 obtain score NFG-4_1000 in Game 0.5329286502805207 
2021-09-11 16:20:13 [I] evolution.py:241 - Algorithm length: 5 ave_score: 0.0 
2021-09-11 16:20:13 [I] evolution.py:241 - Algorithm length: 7 ave_score: 0.6014875204227162 
2021-09-11 16:20:13 [I] evolution.py:241 - Algorithm length: 22 ave_score: 1.0 
2021-09-11 16:20:14 [I] evolution.py:241 - Algorithm length: 7 ave_score: 0.8121523430900276 
2021-09-11 16:20:14 [I] evolution.py:245 - Finish reading and evaluating agents. 
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65570)[0m   return input_values[0] * input_values[1]
2021-09-11 16:20:17 [I] evolution.py:162 - 20 evaluator: waiting: 0, running: 0, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 0 
2021-09-11 16:20:17 [I] evolution.py:165 - 20 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 16 
2021-09-11 16:20:17 [I] evolution.py:168 - 20 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 16, evaluaing: 0, succ: 0, fail: 0, check_fail: 0, drop: 0 
[2m[36m(pid=65627)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65627)[0m policy_prob = -0.103813, 0 = 0
[2m[36m(pid=65629)[0m /home/dell/autocfr/autocfr/program/operations.py:126: RuntimeWarning: invalid value encountered in double_scalars
[2m[36m(pid=65629)[0m   return input_values[0] / input_values[1]
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: overflow encountered in power
[2m[36m(pid=65646)[0m   return np.power(a, b)
[2m[36m(pid=65570)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65570)[0m policy_prob = -0.462442, 0 = 0
[2m[36m(pid=65646)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65646)[0m policy_prob = -0.0704554, 0 = 0
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: invalid value encountered in multiply
[2m[36m(pid=65570)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: overflow encountered in power
[2m[36m(pid=65570)[0m   return np.power(a, b)
2021-09-11 16:20:23 [I] evolution.py:162 - 60 evaluator: waiting: 0, running: 1, finished: 3, removed: 0, finished_removed: 0, error: 0, outputed: 0 
2021-09-11 16:20:23 [I] evolution.py:165 - 60 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 51 
2021-09-11 16:20:23 [I] evolution.py:168 - 60 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 55, evaluaing: 1, succ: 0, fail: 0, check_fail: 0, drop: 0 
[2m[36m(pid=65629)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65629)[0m policy_prob = -0.0390385, 0 = 0
[2m[36m(pid=65571)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65571)[0m policy_prob = -0.0782692, 0 = 0
[2m[36m(pid=65571)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65571)[0m   return np.power(a, b)
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65570)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: invalid value encountered in multiply
[2m[36m(pid=65570)[0m   return input_values[0] * input_values[1]
2021-09-11 16:20:27 [I] evolution.py:162 - 80 evaluator: waiting: 0, running: 1, finished: 3, removed: 0, finished_removed: 0, error: 0, outputed: 4 
2021-09-11 16:20:27 [I] evolution.py:165 - 80 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 75 
2021-09-11 16:20:27 [I] evolution.py:168 - 80 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 74, evaluaing: 1, succ: 1, fail: 0, check_fail: 0, drop: 0 
[2m[36m(pid=65627)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65627)[0m policy_prob = -0.03126, 0 = 0
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65646)[0m   return np.power(a, b)
2021-09-11 16:20:31 [I] evolution.py:162 - 100 evaluator: waiting: 0, running: 0, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 12 
2021-09-11 16:20:31 [I] evolution.py:165 - 100 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 96 
2021-09-11 16:20:31 [I] evolution.py:168 - 100 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 93, evaluaing: 0, succ: 3, fail: 0, check_fail: 0, drop: 0 
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65646)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65629)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65629)[0m policy_prob = -0.00472421, 0 = 0
[2m[36m(pid=65571)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: divide by zero encountered in power
[2m[36m(pid=65571)[0m   return np.power(a, b)
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: overflow encountered in power
[2m[36m(pid=65570)[0m   return np.power(a, b)
[2m[36m(pid=65570)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65570)[0m policy_prob = -2.06821, 0 = 0
2021-09-11 16:20:36 [I] evolution.py:162 - 120 evaluator: waiting: 0, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 16 
2021-09-11 16:20:36 [I] evolution.py:165 - 120 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 116 
2021-09-11 16:20:36 [I] evolution.py:168 - 120 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 111, evaluaing: 1, succ: 4, fail: 0, check_fail: 0, drop: 0 
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65570)[0m   return np.power(a, b)
[2m[36m(pid=65629)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65629)[0m policy_prob = -0.547013, 0 = 0
[2m[36m(pid=65627)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: overflow encountered in power
[2m[36m(pid=65627)[0m   return np.power(a, b)
[2m[36m(pid=65571)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: invalid value encountered in multiply
[2m[36m(pid=65571)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65627)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65627)[0m policy_prob = -0.0598611, 0 = 0
2021-09-11 16:20:40 [I] evolution.py:162 - 140 evaluator: waiting: 0, running: 0, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 20 
2021-09-11 16:20:40 [I] evolution.py:165 - 140 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 139 
2021-09-11 16:20:40 [I] evolution.py:168 - 140 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 131, evaluaing: 0, succ: 5, fail: 0, check_fail: 0, drop: 0 
[2m[36m(pid=65627)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65627)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65570)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65570)[0m policy_prob = -0.505723, 0 = 0
[2m[36m(pid=65170)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65170)[0m   return input_values[0] * input_values[1]
2021-09-11 16:20:42 [W] evolution.py:151 - Program 145 evaluation exception, program name: NFG-1_1000, error info: [<autocfr.program.operations.Multiply object at 0x7fd5b8ad9470>, <autocfr.program.program_types.Vector object at 0x7fd5b8ad9278>, array([ 2.03787904e+003,  9.99295600e-001, -1.57432688e+305]), array([ 9.99821386e-01, -1.78613864e-04,  2.00000015e+04]), array([ 2.03751505e+03, -1.78488049e-04,            -inf])]
Traceback (most recent call last):
  File "/home/dell/autocfr/autocfr/program/executor.py", line 56, in _execute_program
    assert operation.cached_output_type.is_valid_value(output_value)
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/dell/autocfr/autocfr/cfr/cfr_algorithm.py", line 32, in execute
    return self.execute_run(input_values_of_names)
  File "/home/dell/autocfr/autocfr/cfr/cfr_algorithm.py", line 52, in execute_run
    input_values_of_variables, constant_values_of_variables
  File "/home/dell/autocfr/autocfr/program/program.py", line 24, in execute
    data_structure_values,
  File "/home/dell/autocfr/autocfr/program/executor.py", line 64, in _execute_program
    raise ProgramExecutionError(e)
autocfr.program.executor.ProgramExecutionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/dell/autocfr/autocfr/evaluator/evaluator.py", line 62, in evaluate_algorithm
    exp_lower_limit,
  File "/home/dell/autocfr/autocfr/evaluator/evaluator.py", line 33, in evaluate_algorithm_run
    solver.iteration()
  File "/home/dell/autocfr/autocfr/cfr/cfr_solver.py", line 98, in iteration
    s.execute_program(self.algorithm, self.iter_count)
  File "/home/dell/autocfr/autocfr/cfr/cfr_solver.py", line 59, in execute_program
    output_values_of_names = algorithm.execute(input_values_of_names)
  File "/home/dell/autocfr/autocfr/cfr/cfr_algorithm.py", line 36, in execute
    raise executor.AlgorithmExecutionError(e.args[0].info)
autocfr.program.executor.AlgorithmExecutionError: [<autocfr.program.operations.Multiply object at 0x7fd5b8ad9470>, <autocfr.program.program_types.Vector object at 0x7fd5b8ad9278>, array([ 2.03787904e+003,  9.99295600e-001, -1.57432688e+305]), array([ 9.99821386e-01, -1.78613864e-04,  2.00000015e+04]), array([ 2.03751505e+03, -1.78488049e-04,            -inf])]
 
[2m[36m(pid=65627)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65627)[0m   return np.power(a, b)
[2m[36m(pid=65627)[0m /home/dell/autocfr/autocfr/cfr/cfr_solver.py:148: RuntimeWarning: invalid value encountered in double_scalars
[2m[36m(pid=65627)[0m   s.ins_regret[index] += opp_reach * (child_v[a] - v)
[2m[36m(pid=65629)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: overflow encountered in power
[2m[36m(pid=65629)[0m   return np.power(a, b)
2021-09-11 16:20:44 [I] evolution.py:162 - 160 evaluator: waiting: 0, running: 0, finished: 0, removed: 3, finished_removed: 0, error: 1, outputed: 20 
2021-09-11 16:20:44 [I] evolution.py:165 - 160 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 163 
2021-09-11 16:20:44 [I] evolution.py:168 - 160 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 150, evaluaing: 0, succ: 5, fail: 1, check_fail: 0, drop: 0 
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65646)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65629)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65629)[0m policy_prob = -5.11395e-06, 0 = 0
[2m[36m(pid=65629)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65629)[0m policy_prob = -0.0249406, 0 = 0
[2m[36m(pid=65627)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65627)[0m   return np.power(a, b)
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: divide by zero encountered in power
[2m[36m(pid=65570)[0m   return np.power(a, b)
2021-09-11 16:20:49 [I] evolution.py:162 - 180 evaluator: waiting: 4, running: 0, finished: 0, removed: 3, finished_removed: 0, error: 1, outputed: 24 
2021-09-11 16:20:49 [I] evolution.py:165 - 180 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 184 
2021-09-11 16:20:49 [I] evolution.py:168 - 180 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 168, evaluaing: 1, succ: 6, fail: 1, check_fail: 0, drop: 0 
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: overflow encountered in power
[2m[36m(pid=65570)[0m   return np.power(a, b)
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: invalid value encountered in multiply
[2m[36m(pid=65570)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65627)[0m /home/dell/autocfr/autocfr/program/operations.py:222: RuntimeWarning: invalid value encountered in true_divide
[2m[36m(pid=65627)[0m   vec = vec / p_sum
2021-09-11 16:20:52 [I] evolution.py:162 - 200 evaluator: waiting: 0, running: 0, finished: 0, removed: 3, finished_removed: 0, error: 1, outputed: 28 
2021-09-11 16:20:52 [I] evolution.py:165 - 200 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 203 
2021-09-11 16:20:52 [I] evolution.py:168 - 200 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 188, evaluaing: 0, succ: 7, fail: 1, check_fail: 0, drop: 0 
[2m[36m(pid=65629)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65629)[0m   return np.power(a, b)
[2m[36m(pid=65571)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: divide by zero encountered in power
[2m[36m(pid=65571)[0m   return np.power(a, b)
[2m[36m(pid=65629)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65629)[0m   return input_values[0] * input_values[1]
2021-09-11 16:20:56 [I] evolution.py:162 - 220 evaluator: waiting: 0, running: 4, finished: 0, removed: 3, finished_removed: 0, error: 1, outputed: 28 
2021-09-11 16:20:56 [I] evolution.py:165 - 220 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 223 
2021-09-11 16:20:56 [I] evolution.py:168 - 220 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 207, evaluaing: 1, succ: 7, fail: 1, check_fail: 0, drop: 0 
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/cfr/cfr_solver.py:148: RuntimeWarning: invalid value encountered in double_scalars
[2m[36m(pid=65570)[0m   s.ins_regret[index] += opp_reach * (child_v[a] - v)
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: invalid value encountered in multiply
[2m[36m(pid=65570)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65571)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: divide by zero encountered in power
[2m[36m(pid=65571)[0m   return np.power(a, b)
[2m[36m(pid=65570)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65570)[0m policy_prob = -0.0317456, 0 = 0
[2m[36m(pid=65571)[0m /home/dell/autocfr/autocfr/program/operations.py:126: RuntimeWarning: invalid value encountered in true_divide
[2m[36m(pid=65571)[0m   return input_values[0] / input_values[1]
[2m[36m(pid=65629)[0m /home/dell/autocfr/autocfr/cfr/cfr_solver.py:148: RuntimeWarning: invalid value encountered in double_scalars
[2m[36m(pid=65629)[0m   s.ins_regret[index] += opp_reach * (child_v[a] - v)
[2m[36m(pid=65571)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65571)[0m policy_prob = -6.29409e-155, 0 = 0
[2m[36m(pid=65571)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65571)[0m   return np.power(a, b)
2021-09-11 16:21:05 [I] evolution.py:162 - 260 evaluator: waiting: 4, running: 4, finished: 0, removed: 3, finished_removed: 0, error: 1, outputed: 32 
2021-09-11 16:21:05 [I] evolution.py:165 - 260 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 269 
2021-09-11 16:21:05 [I] evolution.py:168 - 260 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 245, evaluaing: 2, succ: 8, fail: 1, check_fail: 0, drop: 0 
[2m[36m(pid=65646)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65646)[0m policy_prob = -0.0265827, 0 = 0
[2m[36m(pid=65629)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65629)[0m policy_prob = -0.00505051, 0 = 0
[2m[36m(pid=65571)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65571)[0m policy_prob = -0.31792, 0 = 0
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65570)[0m   return input_values[0] * input_values[1]
2021-09-11 16:21:10 [I] evolution.py:162 - 280 evaluator: waiting: 0, running: 0, finished: 0, removed: 3, finished_removed: 0, error: 1, outputed: 40 
2021-09-11 16:21:10 [I] evolution.py:165 - 280 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 291 
2021-09-11 16:21:10 [I] evolution.py:168 - 280 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 265, evaluaing: 0, succ: 10, fail: 1, check_fail: 0, drop: 0 
[2m[36m(pid=65629)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: divide by zero encountered in power
[2m[36m(pid=65629)[0m   return np.power(a, b)
2021-09-11 16:21:14 [I] evolution.py:162 - 300 evaluator: waiting: 0, running: 0, finished: 0, removed: 3, finished_removed: 0, error: 1, outputed: 40 
2021-09-11 16:21:14 [I] evolution.py:165 - 300 generator: waiting: 6, running: 3, finished: 1, removed: 0, finished_removed: 0, error: 0, outputed: 310 
2021-09-11 16:21:14 [I] evolution.py:168 - 300 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 285, evaluaing: 0, succ: 10, fail: 1, check_fail: 0, drop: 0 
[2m[36m(pid=65627)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65627)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65627)[0m /home/dell/autocfr/autocfr/program/operations.py:222: RuntimeWarning: invalid value encountered in true_divide
[2m[36m(pid=65627)[0m   vec = vec / p_sum
[2m[36m(pid=65629)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65629)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65627)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65627)[0m   return input_values[0] * input_values[1]
2021-09-11 16:21:18 [I] evolution.py:162 - 320 evaluator: waiting: 0, running: 0, finished: 0, removed: 3, finished_removed: 0, error: 1, outputed: 44 
2021-09-11 16:21:18 [I] evolution.py:165 - 320 generator: waiting: 6, running: 3, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 329 
2021-09-11 16:21:18 [I] evolution.py:168 - 320 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 304, evaluaing: 0, succ: 11, fail: 1, check_fail: 0, drop: 0 
[2m[36m(pid=65571)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: overflow encountered in power
[2m[36m(pid=65571)[0m   return np.power(a, b)
[2m[36m(pid=65571)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65571)[0m   return np.power(a, b)
[2m[36m(pid=65570)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65570)[0m policy_prob = -4.50241e-295, 0 = 0
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:126: RuntimeWarning: invalid value encountered in true_divide
[2m[36m(pid=65646)[0m   return input_values[0] / input_values[1]
2021-09-11 16:21:23 [I] evolution.py:162 - 340 evaluator: waiting: 0, running: 0, finished: 0, removed: 3, finished_removed: 0, error: 1, outputed: 48 
2021-09-11 16:21:23 [I] evolution.py:165 - 340 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 353 
2021-09-11 16:21:23 [I] evolution.py:168 - 340 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 323, evaluaing: 0, succ: 12, fail: 1, check_fail: 0, drop: 0 
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65646)[0m   return np.power(a, b)
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65570)[0m   return np.power(a, b)
[2m[36m(pid=65627)[0m /home/dell/autocfr/autocfr/program/operations.py:126: RuntimeWarning: invalid value encountered in double_scalars
[2m[36m(pid=65627)[0m   return input_values[0] / input_values[1]
[2m[36m(pid=65629)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65629)[0m policy_prob = -5.17508e+14, 0 = 0
2021-09-11 16:21:27 [I] evolution.py:162 - 360 evaluator: waiting: 0, running: 4, finished: 0, removed: 3, finished_removed: 0, error: 1, outputed: 52 
2021-09-11 16:21:27 [I] evolution.py:165 - 360 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 375 
2021-09-11 16:21:27 [I] evolution.py:168 - 360 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 341, evaluaing: 1, succ: 13, fail: 1, check_fail: 0, drop: 0 
[2m[36m(pid=65629)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65629)[0m policy_prob = -0.03126, 0 = 0
[2m[36m(pid=65629)[0m /home/dell/autocfr/autocfr/program/operations.py:126: RuntimeWarning: invalid value encountered in double_scalars
[2m[36m(pid=65629)[0m   return input_values[0] / input_values[1]
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65646)[0m   return np.power(a, b)
[2m[36m(pid=65629)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65629)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65627)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65627)[0m policy_prob = -0.000197379, 0 = 0
[2m[36m(pid=65627)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65627)[0m policy_prob = -0.0135771, 0 = 0
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: invalid value encountered in multiply
[2m[36m(pid=65570)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65627)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65627)[0m   return np.power(a, b)
2021-09-11 16:21:36 [I] evolution.py:162 - 400 evaluator: waiting: 0, running: 0, finished: 0, removed: 3, finished_removed: 0, error: 1, outputed: 60 
2021-09-11 16:21:36 [I] evolution.py:165 - 400 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 416 
2021-09-11 16:21:36 [I] evolution.py:168 - 400 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 380, evaluaing: 0, succ: 15, fail: 1, check_fail: 0, drop: 0 
[2m[36m(pid=65646)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65646)[0m policy_prob = -7.39635, 0 = 0
[2m[36m(pid=65646)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65646)[0m policy_prob = -2.00553e-06, 0 = 0
[2m[36m(pid=65627)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65627)[0m   return np.power(a, b)
[2m[36m(pid=65646)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65646)[0m policy_prob = -0.154296, 0 = 0
[2m[36m(pid=65627)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65627)[0m policy_prob = -1.13174, 0 = 0
[2m[36m(pid=65646)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65646)[0m policy_prob = -7.13186e-05, 0 = 0
2021-09-11 16:21:41 [I] evolution.py:162 - 420 evaluator: waiting: 0, running: 0, finished: 0, removed: 3, finished_removed: 0, error: 1, outputed: 64 
2021-09-11 16:21:41 [I] evolution.py:165 - 420 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 438 
2021-09-11 16:21:41 [I] evolution.py:168 - 420 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 399, evaluaing: 0, succ: 16, fail: 1, check_fail: 0, drop: 0 
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65570)[0m   return np.power(a, b)
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65570)[0m   return np.power(a, b)
[2m[36m(pid=65627)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65627)[0m policy_prob = -8.52545e-05, 0 = 0
2021-09-11 16:21:47 [I] evolution.py:162 - 440 evaluator: waiting: 0, running: 0, finished: 0, removed: 3, finished_removed: 0, error: 1, outputed: 64 
2021-09-11 16:21:47 [I] evolution.py:165 - 440 generator: waiting: 6, running: 3, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 465 
2021-09-11 16:21:47 [I] evolution.py:168 - 440 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 419, evaluaing: 0, succ: 16, fail: 1, check_fail: 0, drop: 0 
[2m[36m(pid=65629)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65629)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65629)[0m /home/dell/autocfr/autocfr/cfr/cfr_solver.py:148: RuntimeWarning: invalid value encountered in double_scalars
[2m[36m(pid=65629)[0m   s.ins_regret[index] += opp_reach * (child_v[a] - v)
[2m[36m(pid=65627)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: divide by zero encountered in power
[2m[36m(pid=65627)[0m   return np.power(a, b)
[2m[36m(pid=65627)[0m /home/dell/autocfr/autocfr/program/operations.py:222: RuntimeWarning: invalid value encountered in true_divide
[2m[36m(pid=65627)[0m   vec = vec / p_sum
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: divide by zero encountered in power
[2m[36m(pid=65646)[0m   return np.power(a, b)
[2m[36m(pid=65627)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65627)[0m policy_prob = -0.711227, 0 = 0
[2m[36m(pid=65571)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: overflow encountered in power
[2m[36m(pid=65571)[0m   return np.power(a, b)
[2m[36m(pid=65570)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65570)[0m policy_prob = -5e-191, 0 = 0
[2m[36m(pid=65627)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65627)[0m policy_prob = -0.363627, 0 = 0
[2m[36m(pid=65629)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65629)[0m policy_prob = -0.00276433, 0 = 0
2021-09-11 16:21:52 [I] evolution.py:162 - 460 evaluator: waiting: 0, running: 5, finished: 3, removed: 3, finished_removed: 0, error: 1, outputed: 64 
2021-09-11 16:21:52 [I] evolution.py:165 - 460 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 484 
2021-09-11 16:21:52 [I] evolution.py:168 - 460 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 437, evaluaing: 2, succ: 16, fail: 1, check_fail: 0, drop: 0 
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: invalid value encountered in multiply
[2m[36m(pid=65570)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65571)[0m /home/dell/autocfr/autocfr/program/operations.py:126: RuntimeWarning: invalid value encountered in double_scalars
[2m[36m(pid=65571)[0m   return input_values[0] / input_values[1]
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65646)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65629)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65629)[0m   return np.power(a, b)
[2m[36m(pid=65571)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65571)[0m policy_prob = -0.00544198, 0 = 0
[2m[36m(pid=65629)[0m /home/dell/autocfr/autocfr/cfr/cfr_solver.py:148: RuntimeWarning: invalid value encountered in double_scalars
[2m[36m(pid=65629)[0m   s.ins_regret[index] += opp_reach * (child_v[a] - v)
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65570)[0m   return np.power(a, b)
2021-09-11 16:21:57 [I] evolution.py:162 - 480 evaluator: waiting: 0, running: 0, finished: 0, removed: 3, finished_removed: 0, error: 1, outputed: 72 
2021-09-11 16:21:57 [I] evolution.py:165 - 480 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 508 
2021-09-11 16:21:57 [I] evolution.py:168 - 480 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 457, evaluaing: 0, succ: 18, fail: 1, check_fail: 0, drop: 0 
[2m[36m(pid=65646)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65646)[0m policy_prob = -0.39846, 0 = 0
[2m[36m(pid=65627)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65627)[0m   return np.power(a, b)
[2m[36m(pid=65646)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65646)[0m policy_prob = -0.193633, 0 = 0
2021-09-11 16:22:01 [I] evolution.py:162 - 500 evaluator: waiting: 0, running: 1, finished: 3, removed: 3, finished_removed: 0, error: 1, outputed: 80 
2021-09-11 16:22:01 [I] evolution.py:165 - 500 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 528 
2021-09-11 16:22:01 [I] evolution.py:168 - 500 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 474, evaluaing: 1, succ: 20, fail: 1, check_fail: 0, drop: 0 
[2m[36m(pid=65629)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: overflow encountered in power
[2m[36m(pid=65629)[0m   return np.power(a, b)
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:126: RuntimeWarning: invalid value encountered in double_scalars
[2m[36m(pid=65646)[0m   return input_values[0] / input_values[1]
[2m[36m(pid=65627)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65627)[0m policy_prob = -5e-286, 0 = 0
2021-09-11 16:22:04 [W] evolution.py:151 - Program 504 evaluation exception, program name: NFG-3_1000, error info: [<autocfr.program.operations.Multiply object at 0x7fd5b8624208>, <autocfr.program.program_types.Vector object at 0x7fd5b8624da0>, array([-3.99965877e-003,  1.14027107e+308]), array([0.        , 1.99991494]), array([-0., inf])]
Traceback (most recent call last):
  File "/home/dell/autocfr/autocfr/program/executor.py", line 56, in _execute_program
    assert operation.cached_output_type.is_valid_value(output_value)
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/dell/autocfr/autocfr/cfr/cfr_algorithm.py", line 32, in execute
    return self.execute_run(input_values_of_names)
  File "/home/dell/autocfr/autocfr/cfr/cfr_algorithm.py", line 52, in execute_run
    input_values_of_variables, constant_values_of_variables
  File "/home/dell/autocfr/autocfr/program/program.py", line 24, in execute
    data_structure_values,
  File "/home/dell/autocfr/autocfr/program/executor.py", line 64, in _execute_program
    raise ProgramExecutionError(e)
autocfr.program.executor.ProgramExecutionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/dell/autocfr/autocfr/evaluator/evaluator.py", line 62, in evaluate_algorithm
    exp_lower_limit,
  File "/home/dell/autocfr/autocfr/evaluator/evaluator.py", line 33, in evaluate_algorithm_run
    solver.iteration()
  File "/home/dell/autocfr/autocfr/cfr/cfr_solver.py", line 98, in iteration
    s.execute_program(self.algorithm, self.iter_count)
  File "/home/dell/autocfr/autocfr/cfr/cfr_solver.py", line 59, in execute_program
    output_values_of_names = algorithm.execute(input_values_of_names)
  File "/home/dell/autocfr/autocfr/cfr/cfr_algorithm.py", line 36, in execute
    raise executor.AlgorithmExecutionError(e.args[0].info)
autocfr.program.executor.AlgorithmExecutionError: [<autocfr.program.operations.Multiply object at 0x7fd5b8624208>, <autocfr.program.program_types.Vector object at 0x7fd5b8624da0>, array([-3.99965877e-003,  1.14027107e+308]), array([0.        , 1.99991494]), array([-0., inf])]
 
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/cfr/cfr_solver.py:148: RuntimeWarning: invalid value encountered in double_scalars
[2m[36m(pid=65570)[0m   s.ins_regret[index] += opp_reach * (child_v[a] - v)
2021-09-11 16:22:06 [I] evolution.py:162 - 520 evaluator: waiting: 0, running: 0, finished: 0, removed: 6, finished_removed: 0, error: 2, outputed: 92 
2021-09-11 16:22:06 [I] evolution.py:165 - 520 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 556 
2021-09-11 16:22:06 [I] evolution.py:168 - 520 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 491, evaluaing: 0, succ: 23, fail: 2, check_fail: 0, drop: 0 
[2m[36m(pid=65646)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65646)[0m policy_prob = -2.24911e-100, 0 = 0
[2m[36m(pid=65571)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: divide by zero encountered in power
[2m[36m(pid=65571)[0m   return np.power(a, b)
2021-09-11 16:22:11 [I] evolution.py:162 - 540 evaluator: waiting: 0, running: 2, finished: 2, removed: 6, finished_removed: 0, error: 2, outputed: 92 
2021-09-11 16:22:11 [I] evolution.py:165 - 540 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 578 
2021-09-11 16:22:11 [I] evolution.py:168 - 540 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 510, evaluaing: 1, succ: 23, fail: 2, check_fail: 0, drop: 0 
2021-09-11 16:22:13 [W] evolution.py:151 - Program 545 evaluation exception, program name: NFG-1_1000, error info: [<autocfr.program.operations.Exp object at 0x7fce2b79ef60>, <autocfr.program.program_types.Vector object at 0x7fce2b7af748>, array([  6666.33333333,   6667.33333333, -13333.66666667]), array([inf, inf,  0.])]
Traceback (most recent call last):
  File "/home/dell/autocfr/autocfr/program/executor.py", line 56, in _execute_program
    assert operation.cached_output_type.is_valid_value(output_value)
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/dell/autocfr/autocfr/cfr/cfr_algorithm.py", line 32, in execute
    return self.execute_run(input_values_of_names)
  File "/home/dell/autocfr/autocfr/cfr/cfr_algorithm.py", line 52, in execute_run
    input_values_of_variables, constant_values_of_variables
  File "/home/dell/autocfr/autocfr/program/program.py", line 24, in execute
    data_structure_values,
  File "/home/dell/autocfr/autocfr/program/executor.py", line 64, in _execute_program
    raise ProgramExecutionError(e)
autocfr.program.executor.ProgramExecutionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/dell/autocfr/autocfr/evaluator/evaluator.py", line 62, in evaluate_algorithm
    exp_lower_limit,
  File "/home/dell/autocfr/autocfr/evaluator/evaluator.py", line 33, in evaluate_algorithm_run
    solver.iteration()
  File "/home/dell/autocfr/autocfr/cfr/cfr_solver.py", line 98, in iteration
    s.execute_program(self.algorithm, self.iter_count)
  File "/home/dell/autocfr/autocfr/cfr/cfr_solver.py", line 59, in execute_program
    output_values_of_names = algorithm.execute(input_values_of_names)
  File "/home/dell/autocfr/autocfr/cfr/cfr_algorithm.py", line 36, in execute
    raise executor.AlgorithmExecutionError(e.args[0].info)
autocfr.program.executor.AlgorithmExecutionError: [<autocfr.program.operations.Exp object at 0x7fce2b79ef60>, <autocfr.program.program_types.Vector object at 0x7fce2b7af748>, array([  6666.33333333,   6667.33333333, -13333.66666667]), array([inf, inf,  0.])]
 
[2m[36m(pid=65570)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65570)[0m policy_prob = -1.97296e-42, 0 = 0
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65646)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65646)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65571)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65571)[0m   return input_values[0] * input_values[1]
2021-09-11 16:22:16 [I] evolution.py:162 - 560 evaluator: waiting: 0, running: 0, finished: 0, removed: 9, finished_removed: 0, error: 3, outputed: 108 
2021-09-11 16:22:16 [I] evolution.py:165 - 560 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 601 
2021-09-11 16:22:16 [I] evolution.py:168 - 560 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 526, evaluaing: 0, succ: 27, fail: 3, check_fail: 0, drop: 0 
[2m[36m(pid=65629)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65629)[0m policy_prob = -0.408094, 0 = 0
[2m[36m(pid=65629)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65629)[0m   return input_values[0] * input_values[1]
2021-09-11 16:22:20 [I] evolution.py:162 - 580 evaluator: waiting: 0, running: 0, finished: 0, removed: 9, finished_removed: 0, error: 3, outputed: 112 
2021-09-11 16:22:20 [I] evolution.py:165 - 580 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 622 
2021-09-11 16:22:20 [I] evolution.py:168 - 580 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 545, evaluaing: 0, succ: 28, fail: 3, check_fail: 0, drop: 0 
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65646)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65571)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65571)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65627)[0m /home/dell/autocfr/autocfr/program/operations.py:126: RuntimeWarning: invalid value encountered in true_divide
[2m[36m(pid=65627)[0m   return input_values[0] / input_values[1]
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: overflow encountered in power
[2m[36m(pid=65646)[0m   return np.power(a, b)
2021-09-11 16:22:23 [I] evolution.py:162 - 600 evaluator: waiting: 0, running: 0, finished: 0, removed: 9, finished_removed: 0, error: 3, outputed: 112 
2021-09-11 16:22:23 [I] evolution.py:165 - 600 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 635 
2021-09-11 16:22:23 [I] evolution.py:168 - 600 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 565, evaluaing: 0, succ: 28, fail: 3, check_fail: 0, drop: 0 
[2m[36m(pid=65629)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65629)[0m policy_prob = -0.193633, 0 = 0
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65646)[0m   return input_values[0] * input_values[1]
2021-09-11 16:22:27 [I] evolution.py:162 - 620 evaluator: waiting: 0, running: 4, finished: 0, removed: 9, finished_removed: 0, error: 3, outputed: 112 
2021-09-11 16:22:27 [I] evolution.py:165 - 620 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 658 
2021-09-11 16:22:27 [I] evolution.py:168 - 620 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 584, evaluaing: 1, succ: 28, fail: 3, check_fail: 0, drop: 0 
[2m[36m(pid=65627)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65627)[0m policy_prob = -0.0065005, 0 = 0
[2m[36m(pid=65627)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65627)[0m policy_prob = -0.022281, 0 = 0
[2m[36m(pid=65627)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65627)[0m policy_prob = -0.0283486, 0 = 0
2021-09-11 16:22:31 [W] evolution.py:151 - Program 631 evaluation exception, program name: NFG-1_1000, error info: [<autocfr.program.operations.Exp object at 0x7f5336f9b470>, <autocfr.program.program_types.Vector object at 0x7f5337336940>, array([  6666.33333333,   6667.33333333, -13333.66666667]), array([inf, inf,  0.])]
Traceback (most recent call last):
  File "/home/dell/autocfr/autocfr/program/executor.py", line 56, in _execute_program
    assert operation.cached_output_type.is_valid_value(output_value)
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/dell/autocfr/autocfr/cfr/cfr_algorithm.py", line 32, in execute
    return self.execute_run(input_values_of_names)
  File "/home/dell/autocfr/autocfr/cfr/cfr_algorithm.py", line 52, in execute_run
    input_values_of_variables, constant_values_of_variables
  File "/home/dell/autocfr/autocfr/program/program.py", line 24, in execute
    data_structure_values,
  File "/home/dell/autocfr/autocfr/program/executor.py", line 64, in _execute_program
    raise ProgramExecutionError(e)
autocfr.program.executor.ProgramExecutionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/dell/autocfr/autocfr/evaluator/evaluator.py", line 62, in evaluate_algorithm
    exp_lower_limit,
  File "/home/dell/autocfr/autocfr/evaluator/evaluator.py", line 33, in evaluate_algorithm_run
    solver.iteration()
  File "/home/dell/autocfr/autocfr/cfr/cfr_solver.py", line 98, in iteration
    s.execute_program(self.algorithm, self.iter_count)
  File "/home/dell/autocfr/autocfr/cfr/cfr_solver.py", line 59, in execute_program
    output_values_of_names = algorithm.execute(input_values_of_names)
  File "/home/dell/autocfr/autocfr/cfr/cfr_algorithm.py", line 36, in execute
    raise executor.AlgorithmExecutionError(e.args[0].info)
autocfr.program.executor.AlgorithmExecutionError: [<autocfr.program.operations.Exp object at 0x7f5336f9b470>, <autocfr.program.program_types.Vector object at 0x7f5337336940>, array([  6666.33333333,   6667.33333333, -13333.66666667]), array([inf, inf,  0.])]
 
[2m[36m(pid=65629)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65629)[0m policy_prob = -2.50047e-287, 0 = 0
[2m[36m(pid=65571)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65571)[0m   return np.power(a, b)
[2m[36m(pid=65627)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65627)[0m policy_prob = -0.184749, 0 = 0
2021-09-11 16:22:36 [I] evolution.py:162 - 660 evaluator: waiting: 0, running: 0, finished: 0, removed: 12, finished_removed: 0, error: 4, outputed: 116 
2021-09-11 16:22:36 [I] evolution.py:165 - 660 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 699 
2021-09-11 16:22:36 [I] evolution.py:168 - 660 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 623, evaluaing: 0, succ: 29, fail: 4, check_fail: 0, drop: 0 
[2m[36m(pid=65629)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65629)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65646)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: invalid value encountered in power
[2m[36m(pid=65646)[0m   return np.power(a, b)
[2m[36m(pid=65570)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65570)[0m policy_prob = -0.0065005, 0 = 0
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65646)[0m   return input_values[0] * input_values[1]
[2m[36m(pid=65646)[0m /home/dell/autocfr/autocfr/program/operations.py:155: RuntimeWarning: overflow encountered in power
[2m[36m(pid=65646)[0m   return np.power(a, b)
2021-09-11 16:22:40 [W] evolution.py:151 - Program 672 evaluation exception, program name: NFG-1_1000, error info: [<autocfr.program.operations.Exp object at 0x7fe4f9242d30>, <autocfr.program.program_types.Vector object at 0x7fe4f9242dd8>, array([  6666.33333333,   6667.33333333, -13333.66666667]), array([inf, inf,  0.])]
Traceback (most recent call last):
  File "/home/dell/autocfr/autocfr/program/executor.py", line 56, in _execute_program
    assert operation.cached_output_type.is_valid_value(output_value)
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/dell/autocfr/autocfr/cfr/cfr_algorithm.py", line 32, in execute
    return self.execute_run(input_values_of_names)
  File "/home/dell/autocfr/autocfr/cfr/cfr_algorithm.py", line 52, in execute_run
    input_values_of_variables, constant_values_of_variables
  File "/home/dell/autocfr/autocfr/program/program.py", line 24, in execute
    data_structure_values,
  File "/home/dell/autocfr/autocfr/program/executor.py", line 64, in _execute_program
    raise ProgramExecutionError(e)
autocfr.program.executor.ProgramExecutionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/dell/autocfr/autocfr/evaluator/evaluator.py", line 62, in evaluate_algorithm
    exp_lower_limit,
  File "/home/dell/autocfr/autocfr/evaluator/evaluator.py", line 33, in evaluate_algorithm_run
    solver.iteration()
  File "/home/dell/autocfr/autocfr/cfr/cfr_solver.py", line 98, in iteration
    s.execute_program(self.algorithm, self.iter_count)
  File "/home/dell/autocfr/autocfr/cfr/cfr_solver.py", line 59, in execute_program
    output_values_of_names = algorithm.execute(input_values_of_names)
  File "/home/dell/autocfr/autocfr/cfr/cfr_algorithm.py", line 36, in execute
    raise executor.AlgorithmExecutionError(e.args[0].info)
autocfr.program.executor.AlgorithmExecutionError: [<autocfr.program.operations.Exp object at 0x7fe4f9242d30>, <autocfr.program.program_types.Vector object at 0x7fe4f9242dd8>, array([  6666.33333333,   6667.33333333, -13333.66666667]), array([inf, inf,  0.])]
 
[2m[36m(pid=65570)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65570)[0m policy_prob = -5.00005e-96, 0 = 0
[2m[36m(pid=65629)[0m /home/dell/autocfr/autocfr/program/operations.py:126: RuntimeWarning: invalid value encountered in double_scalars
[2m[36m(pid=65629)[0m   return input_values[0] / input_values[1]
[2m[36m(pid=65570)[0m /home/dell/autocfr/autocfr/program/operations.py:112: RuntimeWarning: overflow encountered in multiply
[2m[36m(pid=65570)[0m   return input_values[0] * input_values[1]
2021-09-11 16:22:42 [I] evolution.py:162 - 680 evaluator: waiting: 0, running: 0, finished: 0, removed: 15, finished_removed: 0, error: 5, outputed: 124 
2021-09-11 16:22:42 [I] evolution.py:165 - 680 generator: waiting: 6, running: 4, finished: 0, removed: 0, finished_removed: 0, error: 0, outputed: 730 
2021-09-11 16:22:42 [I] evolution.py:168 - 680 Agent Counter: generating: 0, early_hurdle: 0, func_equiv: 640, evaluaing: 0, succ: 31, fail: 5, check_fail: 0, drop: 0 
[2m[36m(pid=65629)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65629)[0m policy_prob = -0.391985, 0 = 0
[2m[36m(pid=65646)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65646)[0m policy_prob = -3.20485e+06, 0 = 0
[2m[36m(pid=65646)[0m OpenSpiel exception: /tmp/pip-req-build-2kcuwycz/open_spiel/algorithms/history_tree.cc:215 policy_prob >= 0
[2m[36m(pid=65646)[0m policy_prob = -0.494817, 0 = 0
